### PR TITLE
feat(CoverallsBadge): add badge and test

### LINKED ISSUE #3: Add a CoverallsBadge component
Some repositories use [Coveralls](https://coveralls.io/) for test coverage. A badge showing the current test coverage helps to understand whether the repo currently is in good shape.

This issue is meant to be a good first issue to get into this repository, and into open source contribution over all, as it will require to touch few existing files and is documented extensively. If this is your first contribution to open source, I encourage you to try this issue. Please note that I will only accept one of the good first issues per person to give as many people as possible the opportunity for a first contribution.

Please read our [contribution guidelines](https://github.com/dbartholomae/jsx-readme/blob/main/CONTRIBUTING.md) before picking this up, it will also lead you through the process of setting up the repository.

## Describe the solution you'd like

This is what the badge should look like in its most basic form:

[code coverage](https://coveralls.io/github/dbartholomae/jsx-readme?branch=main)

It should be accessible as `<CoverallsBadge pkg={pkg} />`, with an optionals props `branch` that can be used as `<CoverallsBadge pkg={pkg} branch='master' />` to render to:

[code coverage](https://coveralls.io/github/dbartholomae/jsx-readme?branch=master)

The badge needs to be exported from the [badges index barrel file](https://github.com/dbartholomae/jsx-readme/blob/main/src/components/badges/index.ts).
Please note that you will also have to provide test coverage and documentation for me to be able to accept the PR.

If you are interested in an additional challenge, try to also support BitBucket and Gitlab in addition to GitHub, based on package.json

## Additional context

A good badge to base this new badge on is the [GitHub Workflow badge](https://github.com/dbartholomae/jsx-readme/blob/main/src/components/badges/GithubWorkflowBadge.tsx).

### PR DESCRIPTION
**Pull request by MDShields7**
Closes #3 
## Checklist:
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by MDShields7** [comment]
I had these errors earlier locally, which stem from my labelling a file as Javascript. I accidentally named the file 'CoverallsBadge.js' to start, and later changed it to 'CoverallsBadge.tsx'. However, the CoverallsBadge.test.tsx was still importing from "./CoverallsBadge" (CoverallsBadge being a .js file in its mind).
I fixed this issue locally, and my pre-commit tests passed, but it seems the error has travelled to the build somehow. Since my local version seems fine, I am not sure how to fix this, except by trial and error.
Do you have a suggestion?

**Comment by dbartholomae** [comment]
Thanks! In the commit there is a file "src/components/badges/CoverallsBadge.js" which should be called " src/components/badges/CoverallsBadge.tsx"

File: src/components/badges/CoverallsBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  if (branch === undefined) {
+    return (
+      <Badge
+        link={`https://coveralls.io/github/${owner}/${repo}`}
+        imageSource={`https://coveralls.io/repos/github/${owner}/${repo}/badge.svg`}
+      >
+        coveralls
+      </Badge>
+    );
+  } else {
```
Comment (dbartholomae) [inline src/components/badges/CoverallsBadge.tsx:30]: No need for the `else` :)

**Review by dbartholomae — CHANGES_REQUESTED** [review]
PR looks mostly fine. One small stylistic change, and you'll have to export the badge from the barrel file src/components/badges/index.ts so that it can be used.

**Review by dbartholomae — APPROVED** [review]

**Comment by dbartholomae** [comment]
Thanks, looks great! :)

**Pull request merged**